Refactor TimML plotting

dbrakenhoff opened the pull request:
- Create plotting submodule, `ml.plots`
- Add deprecation warnings for old plotting methods, and use new syntax.
- Add cross-section plot (adapted from TTim), add cross-section plotting to some 1D elements, add hstar
- Fix changetrace for XsectionAreaSink element
- Remove changetrace from XsectionAreaSinkInhom element

mbakker7 commented:
Excellent start. What I think that should be added (similar to `ttim`):
1. Gather plotting the xsection in a function `ml.plots.xsection`
2. Add vertical dotted line between sections, as in ttim
3. Add hstar (in 'C0' blue) in case of semi-confined flow.

mbakker7 commented:
What probably should be fixed:
1. layout=True in Vcontour should probably plot the xsection.
2. Some of the boundaries between xsections don't go to the right elevation
3. What about drawing xsection elements in the xsection. Is that difficult?

dbrakenhoff commented:
> 1. layout=True in Vcontour should probably plot the xsection.

Agreed.

> 2. Some of the boundaries between xsections don't go to the right elevation

Fixed in my latest commit.

> 3. What about drawing xsection elements in the xsection. Is that difficult?

It actually already does so, but only if the original model is a ModelXsection. This is because you can also draw cross-sections of the 3D Models,  but since there is no logic to compute projected distances of elements onto the cross-section line, I decided to skip plotting elements altogether in those cases. 

If you have a 1D element, those have a `plot` method that you can use separately. LeakyLineDoublets did not yet have a plot method, so I added those as well.
